docs/cloudformation/federation.template.yaml

```diff
@@ -0,0 +1,60 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: Janus integration for login federation
+
+# !! This template is provided as an example and is not intended for production use !!
+--!! This line invalidates the template and will need to be removed before CloudFormation !!--
+
+Parameters:
+  JanusInstanceRoleArn:
+    Description: ARN of Janus' IAM role
+    Type: String
+    AllowedPattern: "arn:aws:iam::[0-9]{12}:.+"
+    ConstraintDescription: Should be an AWS IAM role
+  JanusDevUserArn:
+    Description: User for developer testing
+    Type: String
+    AllowedPattern: "^$|arn:aws:iam::[0-9]{12}:.+"
+    ConstraintDescription: Optional ARN of an AWS IAM user
+
+Resources:
+  # Janus uses `assumeRole` to do login federation and needs a role in each target account
+  # https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_request.html#stsapi_comparison
+  FederationRole:
+    Type: AWS::IAM::Role
+    Properties:
+      Path: /federation/
+      MaxSessionDuration: 36000
+      AssumeRolePolicyDocument:
+        Statement:
+        - Effect: Allow
+          Principal:
+            AWS:
+            - !Ref JanusInstanceRoleArn
+            - !If [ HasDevUser, !Ref JanusDevUserArn, !Ref 'AWS::NoValue' ]
```

Comment:
This is neat, but it looks like this condition (`HasDevUser) is not defined anywhere. The documentation tells me we'd need to add a `Conditions` section to the top level of the template that includes a `HasDevUser` entry. In practice, it may also be simpler to invert this Boolean so we can compare to the empty string to define the condition without also using `!Not`? No harm though, this is what `!Not` is for.

    Conditions:
      HasDevUser:
        !Not !Equals [ !Ref JanusDevUserArn, "" ]

(or inverted to avoid the `!Not` with a change to this `!If` statement if so)

This probably needs testing, I've not used conditions before so be good to see it working in AWS.

Reply:
Good catch, I must have missed out committing the Conditions 🤦 
I am removing the idea of the Dev User for now as I agree with your points around not making it the default approach.